pkg/client/unversioned/helper.go

```diff
@@ -460,32 +465,25 @@ func DefaultServerURL(host, prefix string, groupVersion unversioned.GroupVersion
 		}
 		hostURL, err = url.Parse(scheme + base)
 		if err != nil {
-			return nil, err
+			return nil, "", err
 		}
 		if hostURL.Path != "" && hostURL.Path != "/" {
-			return nil, fmt.Errorf("host must be a URL or a host:port pair: %q", base)
+			return nil, "", fmt.Errorf("host must be a URL or a host:port pair: %q", base)
 		}
 	}
 
-	// If the user specified a URL without a path component (http://server.com), automatically
-	// append the default prefix
-	if hostURL.Path == "" {
-		if prefix == "" {
-			prefix = "/"
-		}
-		hostURL.Path = prefix
-	}
+	versionedAPIPath := path.Join("/", hostURL.Path, apiPath)
```

Comment:
This is confusing to me. I would expect `hostURL.Path` to be empty since it should just be the host. What does it mean if `host` is "example.com/prefix_one/" and `apiPath` is "prefix_two"?